src/main/java/com/powsybl/openloadflow/sensi/OpenSensitivityAnalysisProvider.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.sensi;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.math.matrix.MatrixFactory;
+import com.powsybl.openloadflow.OpenLoadFlowParameters;
+import com.powsybl.sensitivity.*;
+import com.powsybl.tools.PowsyblCoreVersion;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+@AutoService(SensitivityAnalysisProvider.class)
+public class OpenSensitivityAnalysisProvider implements SensitivityAnalysisProvider {
+
+    private static final String NAME = "OpenSensitivityAnalysis";
+
+    private final DcSensitivityAnalysis dcSensitivityAnalysis;
+
+    private final AcSensitivityAnalysis acSensitivityAnalysis;
+
+    public OpenSensitivityAnalysisProvider(MatrixFactory matrixFactory) {
```

Comment:
Don't we also need an empty constructor 
`public OpenSensitivityAnalysisProvider()` ? Otherwise the service loader (called by PlatformConfigNamedProvider) would throw 
` java.lang.NoSuchMethodException: com.powsybl.openloadflow.sensi.OpenSensitivityAnalysisProvider.<init>()` when trying to load the Service. 
Or is there a way to provide the matrixFactory to the service loader ?

Reply:
indeed, we do need to add such a constructor, good catch! we should add an empty constructor which takes `SparseMatrixFactory` as default matrix factory.